### PR TITLE
Feature/blog activity category

### PR DESCRIPTION
**Pull request by Ponchimeow**

### CODE REVIEW

**Comment by Ponchimeow** [comment]
[(2) 【分類】部落格分類獨立在文章區、活動分類獨立顯示](https://trello.com/c/GlF8Cyyg/2264-2-%E3%80%90%E5%88%86%E9%A1%9E%E3%80%91%E9%83%A8%E8%90%BD%E6%A0%BC%E5%88%86%E9%A1%9E%E7%8D%A8%E7%AB%8B%E5%9C%A8%E6%96%87%E7%AB%A0%E5%8D%80%E3%80%81%E6%B4%BB%E5%8B%95%E5%88%86%E9%A1%9E%E7%8D%A8%E7%AB%8B%E9%A1%AF%E7%A4%BA)

File: src/pages/BlogPostPage.tsx [hunk 1]
```diff
@@ -192,7 +192,7 @@ const BlogPostPage: React.VFC = () => {
             </div>
             <div className="mb-5">
               {post?.tags.map(tag => (
-                <Link key={tag} to={`/posts/?tags=${tag}`} className="mr-2">
+                <Link key={tag} to={`/posts/?categories=${tag}`} className="mr-2">
```
Comment (kkshyu) [inline src/pages/BlogPostPage.tsx:195]: Do not need to change this one. we should support two kinds of query params: tags and categories
not change all the tags to categories

File: src/pages/BlogPostCollectionPage.tsx [hunk 2]
```diff
@@ -52,9 +52,9 @@ const StyledAbstract = styled.div`
 `
 
 const BlogPostCollectionPage: React.VFC = () => {
-  const [tags] = useQueryParam('tags', StringParam)
+  const [categories] = useQueryParam('categories', StringParam)
```
Comment (kkshyu) [inline src/pages/BlogPostCollectionPage.tsx:55]: Do not need to change this one. we should support two kinds of query params: tags and categories
not change all the tags to categories

**Review by kkshyu — CHANGES_REQUESTED** [review]

File: src/hooks/blog.ts [hunk 3]
```diff
@@ -60,8 +60,8 @@ export const usePostPreviewCollection = (filter?: { authorId?: string; tags?: st
           .filter(
             post =>
               !filter ||
-              typeof filter.tags === 'undefined' ||
-              post.post_tags.some(postTag => filter.tags?.includes(postTag.tag_name)),
+              typeof filter.categories === 'undefined' ||
+              post.post_tags.some(postTag => filter.categories?.includes(postTag.tag_name)),
```
Comment (kkshyu) [inline src/hooks/blog.ts:64]: if `tags` were set, need to filter, too.

File: src/pages/BlogPostCollectionPage.tsx [hunk 4]
```diff
@@ -69,11 +70,19 @@ const BlogPostCollectionPage: React.VFC = () => {
...
+                <span key={tag} className="ml-2">
+                  #{tag}
+                </span>
+              ))}
+            {categories &&
+              categories?.split(',').map(category => (
+                <span key={category} className="ml-2">
+                  {category}
+                </span>
+              ))}
```
Comment (kkshyu) [inline src/pages/BlogPostCollectionPage.tsx:85]: there is optional chaining already, don't need to have the condition

```ts
{
    !categories && tags?.split(',').map(tag => (
                <span key={tag} className="ml-2">
                  #{tag}
                </span>
              ))
}
```

**Review by kkshyu — APPROVED** [review]

**Pull request merged**